src/rpc/claimtrie.cpp

```diff
@@ -35,12 +33,43 @@ static CBlockIndex* BlockHashIndex(const uint256& blockHash)
     if (!chainActive.Contains(pblockIndex))
         throw JSONRPCError(RPC_INTERNAL_ERROR, "Block not in main chain");
 
+    const size_t currentMemoryUsage = pcoinsTip->DynamicMemoryUsage();
+    const int MAX_RPC_BLOCK_DECREMENTS = nCoinCacheUsage / currentMemoryUsage;
```

Comment:
I like the idea of limiting decrements on RAM. However, I don't think this line accomplishes that. You can't make an assumption that each block uses the same amount of coin cache. I think this check of MAX decrements could probably go away. We just use available memory then give up.

Reply:
It *relaxing* check if it want some deeper block that is *easy* knowing that rollback will fail.

Reply:
We don't know that 100 blocks back will fail. It's dependent upon the number of TXOs in each block.

Reply:
Yep but most likely will fail, i think we need some value that will prevent rollback to take time for nothing. nCoinCacheUsage / currentMemoryUsage * 1.5 or something ?

Reply:
I agree with Brannon that this approach

'MAX_RPC_BLOCK_DECREMENTS = nCoinCacheUsage / currentMemoryUsage' is not correct. 

because current memory usage is not predictive of what memory usage a roll back will cost. 

I'm in favor of keeping MAX_RPC_BLOCK_DECREMENTS to a high constant. I think the previous max of 50 is fine, or we could divide nCoinCacheUsage by 20 mb (in my tests I saw that a 1mb block used about 10 mb of cache). This is  in case someone accidentally sets a very high roll back, we don't have to waste time or resource trying to roll back, we will know immediately that the roll back is too high.